src/data/dfinsupp.lean

```diff
@@ -313,6 +313,28 @@ by simp only [single_apply, dif_neg h]
 lemma single_injective {i} : function.injective (single i : β i → Π₀ i, β i) :=
 λ x y H, congr_fun (mk_injective _ H) ⟨i, by simp⟩
 
+/-- Like `finsupp.single_eq_single_iff`, but with a `heq` due to dependent types -/
+lemma single_eq_single_iff (i j : ι) (xi : β i) (xj : β j) :
+  dfinsupp.single i xi = dfinsupp.single j xj ↔ i = j ∧ xi == xj ∨ xi = 0 ∧ xj = 0 :=
```

Comment:
For a lemma like this I tend to also want a lemma of the form `i = j → xi == xj → dfinsupp.single i xi = dfinsupp.single j xj` (the most common case that `dfinsupp.single` are equal)

Reply:
I found that if I want that case, I can just use `congr`. Is the lemma form still worthwhile?